src/data/complex/is_R_or_C.lean

```diff
@@ -710,6 +703,100 @@ by simp [is_R_or_C.abs, abs, real.sqrt_mul_self_eq_abs]
 
 end cleanup_lemmas
 
+section linear_maps
+
+variables {K : Type*} [is_R_or_C K]
+
+/-- The real part in a `is_R_or_C` field, as a linear map. -/
+noncomputable def re_lm : K →ₗ[ℝ] ℝ :=
+{ map_smul' := smul_re',  .. re }
```

Comment:
Since you are using the naming convention `is_R_or_C.re_lm` for the linear-map versions (which I think is fine), could you rename the complex versions?  Currently they are named `complex.linear_map.re`, etc., and live in `data.complex.module`.

Reply:
I've changed the names in `data.complex.module`. Now let's wait for build errors, to see where else those names were used.

Reply:
This change will conflict with #6562, which renamed `smul_re'` to `smul_re`.